src/processors/ipfsFile/mediaUploader.ts

```diff
@@ -0,0 +1,96 @@
+import { urlSource } from '../../clients/ipfs';
+import { Table } from '../../db/db';
+import { IPFSFile, IPFSFileUrl } from '../../types/ipfsFile';
+import { Clients, Processor } from '../../types/processor';
+import { ProcessedTrack } from '../../types/track';
+import { Trigger } from '../../types/trigger';
+import { rollPromises } from '../../utils/rollingPromises';
+
+type TrackUrlJoin = ProcessedTrack & Partial<IPFSFileUrl>;
+type MediaType = 'Artwork' | 'Audio';
+
+const notOnIpfs: (media: MediaType) => Trigger<undefined> =
+(media) => async (clients: Clients) => {
+  const query = `select * from "${Table.processedTracks}" as t
```

Comment:
can we select just the fields we need, instead of * ?

Reply:
I guess we may need something like `idFields: string[],` in the above RecordUrl too so that we know which field to select for the record id(s)